FogCreek/Glitch-Community
# Spectrum splash: rebuild AddCollectionProjectPop

## Links
* Remix link: https://spectrum-splash.glitch.me
* Manuscript link: https://glitch.manuscript.com/f/cases/3327873, https://glitch.manuscript.com/f/cases/3328551/,
https://glitch.manuscript.com/f/cases/3328531/

## GIF/Screenshots
![Screen Shot 2019-05-22 at 5 25 43 PM](https://user-images.githubusercontent.com/938722/58210924-893fd780-7cb8-11e9-9a29-6ec4a750b62e.png)
![Screen Shot 2019-05-22 at 5 26 03 PM](https://user-images.githubusercontent.com/938722/58210925-893fd780-7cb8-11e9-89f0-38a3fca2f8cb.png)
![Screen Shot 2019-05-22 at 5 26 59 PM](https://user-images.githubusercontent.com/938722/58210927-893fd780-7cb8-11e9-8935-f835ebb9781a.png)
![Screen Shot 2019-05-23 at 10 51 06 AM](https://user-images.githubusercontent.com/938722/58262690-ba67e880-7d48-11e9-830c-85010d0ce895.png)

## Changes:
* use Algolia search API
* add filterTypes to algolia search
* componentize AddCollectionProjectPop
* componentize ProjectResultItem 
* add configurable size to ProfileList
* make ProjectResultItem load its own user list, and only when visible
* make results selectable with arrow keys
* fix tooltip overlaps with private projects

## How To Test:
* log into https://spectrum-splash.glitch.me (with email login or using the console)
* create a collection
* clicking "Add Project" should show you a list of your recent projects
* using the up & down arrow keys should highlight projects and scroll them into view
* pressing enter on a highlighted project, or clicking on one, should add it to your collection
* typing in the search box should search the whole community for projects
* you should also be able to paste in URLs of projects to find them
* it should tell you if you attempt to add a project that is already in the collection

## Feedback I'm looking for
* does this feel responsive?
* does it find the projects you'd expect to find?
* should we get rid of the "excluded N search results" warning and just show that a search result is already in the collection?
